### PR TITLE
New submission flow: Your consent screen (EXPOSUREAPP-3735)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Your consent screen.
Navigation would be placed in separate PR.

Your-Consent-Screentshot

### CODE REVIEW

**Review by Oliver-Zimmerman — DISMISSED** [review]
Looks good to me 👍 

Obviously, we need to finalize the flow in terms of navigation once the submission PRs are merged. Then we can test how they all work together. As a single PR though it looks good to me.

File: Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            app:layout_constraintVertical_bias="1.0">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:focusable="true">
+
+                <include
+                    android:id="@+id/submission_your_consent_switch"
+                    layout="@layout/include_settings_switch_row"
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml:42]: Not sure if this should be changed here but the figma design has the switch aligned with the header. For us it looks like it sitting in the middle of the header and the body.

File: Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_marginTop="@dimen/spacing_normal"
+                        android:text="@string/submission_your_consent_agreement_share_symptoms"/>
+
+                </LinearLayout>
+
+                <View
+                    android:id="@+id/submission_your_consent_agreement_details_divider_top"
+                    android:layout_width="@dimen/match_constraint"
+                    android:layout_height="@dimen/card_divider"
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml:113]: For whatever reason this divider and the bottom one have 1.48px height in the figma design. Now that I look at it, all  of the dividing lines from the design have different heights...

File: Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="@dimen/spacing_small"
+                    android:text="@string/submission_your_consent_agreement_details"
+                    app:layout_constraintEnd_toEndOf="@id/guideline_end"
+                    app:layout_constraintStart_toStartOf="@id/guideline_start"
+                    app:layout_constraintTop_toBottomOf="@+id/submission_your_consent_agreement_details_divider_top" />
+
+                <View
+                    android:layout_width="@dimen/match_constraint"
+                    android:layout_height="@dimen/card_divider"
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/res/layout/fragment_submission_your_consent.xml:133]: Same thing as with the top divider.

**Comment by SamuraiKek** [comment]
This looks good to me. There are some minor design differences to figma but they are includes that were already added in a different PR. I'm not sure if these should be changed here so once all your checks pass I'll approve it.

**Review by Oliver-Zimmerman — DISMISSED** [review]
Re-review after lint changes. LGTM 👍

**Review by SamuraiKek — DISMISSED** [review]

**Review by Oliver-Zimmerman — APPROVED** [review]

**Review by SamuraiKek — APPROVED** [review]

**Pull request merged**